core/includes/batch.inc

```diff
@@ -453,7 +453,7 @@ function _batch_finished() {
 
   // Clean-up the session. Not needed for CLI updates.
   if (isset($_SESSION)) {
-    unset($_SESSION['batches'][$batch['id']]);
+    unset($_SESSION['batches'][$_batch['id']]);
```

Comment:
Nice bug fix here. If `$batch` is set to NULL just three lines above, there's no way we can check `$batch['id']`.